### PR TITLE
Allow Booleans as permissible return values from YML

### PR DESCRIPTION
**Pull request by smoll**
Given a local.yml that looks like this:

```
    base_url:  http://system_test.mycompany.com
    use_ssl: false
```

This will allow us to call:
```ruby
FigNewton.use_ssl
# => false
```

Without this change we get the following:
```ruby
FigNewton.use_ssl
# => undefined method `use_ssl' for #<FigNewton::Node:0x007fbd754c9408>
```

### CODE REVIEW

**Comment by smoll** [comment]
@cheezy let me know what you think. I can also add a couple cucumber tests for this if you'd like.

File: features/fig_newton.feature [hunk 1]
```diff
@@ -57,18 +64,28 @@ Feature: Functionality of the fig_newton gem
     Given I have read the configuration file
     When I ask for a value that does not exist named "does_not_exist" that has a default value "the default value"
     Then I should see "the default value"
-  
+
+  Scenario: Requesting data that does not exist but has a default value of true should return the default value of true
```
Comment (smoll) [inline features/fig_newton.feature]: This scenario and the one below it are failing in local tests -- looks like my proposed change does not work for booleans specified as defaults. Not sure if this is even necessary, but looking into it.

File: lib/fig_newton/missing.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ def method_missing(*args, &block)
       read_file unless @yml
       m = args.first
       value = @yml[m.to_s]
+      return value if type_bool?(value)
```
Comment (smoll) [inline lib/fig_newton/missing.rb]: I have an alternate solution that allows the tests to pass but raises the cyclomatic complexity of the method to unacceptable levels according to RuboCop: https://github.com/smoll/fig_newton/commit/2f5e3cc264db9e60527f467bcd96b3618ca1fb26#diff-27da7be4f06451c659e80246180dbfb7L12

Not really sure how to better rework this, though. I'm open to suggestions if anyone has any.

**Comment by stevenjackson** [comment]
@smoll Check out #8.  See if that works for you.

**Pull request closed without merging**